### DISCUSSION TITLE
How to plot all covariates even when 1 is missing to avoid redundancy

### DISCUSSION THREAD

**Discussion by mapapepa** [CamDavidsonPilon/lifelines#1291] [Q&A] [answered]
HI all,
I'm trying to make a plot of the covariates. I read in the documentation that Is should use plot_partial_effects_on_outcome however I can't upgrade my python version so I can only use plot_covariate_groups.  I have 2 questions:
- Is there any documentation of plot_covariate_group? I'm not sure about the arguments I need to use
- I followed an example and managed to produce a graph with plot_covariate_group, however it contains all covariates -1, as I had to discard one to avoid redundancy. If I try to include all groups I get an error saying 'covariate `X` is not present in the original dataset', so I was wondering if there is a way to include it in the graph?

**Comment by CamDavidsonPilon** [comment, marked as answer]
Hi @mapapepa,

1. The docs should be very close to `plot_partial_effects_on_outcome`, however, you can inspect the source code for that function here: https://github.com/CamDavidsonPilon/lifelines/tree/v0.24.16. I can be more specific with where to look, but I need to know what model you are using. 
2. `however it contains all covariates -1` I'm not sure what you mean by `-1`... is that a formula? It would help if you could post the code for `plot_covariate_groups` and the full error message you see.

**Reply by mapapepa** [reply]
Hi @CamDavidsonPilon , thanks for your reply.
1. I'm using a Cox Model.
2. I'm working on a churn analysis, and I basically have a column that describes the type of contract. This column can take up to 5 values. I processed the original dataframe and encoded this variable (creating k-1 dummies), it means that I ended up with 4 columns in the data frame I use for fitting the model.

To plot the covariates I'm using:

`cph.plot_covariate_groups(['membership_duration_P1M', 'membership_duration_P2M', 'membership_duration_P4M',
                           'membership_duration_P6M'], np.eye(4))
plt.xlabel("Month")
plt.ylabel("Prob. of Survival")`

And this contains the 4 dummies, however, I would like to see all types of contracts.  But then if I try to call all of them I get:
''covariate `membership_duration_P12M` is not present in the original dataset'', which makes sense. The question then is, how can I make it work? I was checking your churn example in Github and there I see that for the regression a dummy is taken but the plot is complete so I'm a bit confused about how you did that.

**Reply by CamDavidsonPilon** [reply]
ah I understand. Is there a "baseline" plot also added? That _should_ represent the final column (implicitly), HOWEVER, I did change what the baseline represented for categorical variables recently. Can you tell me what version of lifelines you are using?

```
import lifelines
lifelines.__version__
```

**Reply by mapapepa** [reply]
Yes, there is a baseline plot added but it looks the average though. 
I'm using version 0.23.9

**Reply by CamDavidsonPilon** [reply]
Okay, try something like:

```python
values = np.eye(5)[:, :-1]

cph.plot_covariate_groups(['membership_duration_P1M', 'membership_duration_P2M', 'membership_duration_P4M', 'membership_duration_P6M'], values) plt.xlabel("Month") plt.ylabel("Prob. of Survival")
```

**Reply by mapapepa** [reply]
worked like a charm. Thanks a lot for your help :)
